haicrypt/hcrypt_msg.h

```diff
@@ -72,6 +71,7 @@ typedef struct {
 
 #define hcryptMsg_PaddedLen(len, fact)  ((((len)+(fact)-1)/(fact))*(fact))
 
+int hcryptMsg_SRT_ParseMsg(unsigned char* msg);
```

Comment:
Well, why only this one? If it is ok to have this one function exported, why not all others as well?

And if this generic form of the solution was legit, maybe this function should remain "virtual"?

Reply:
To minimize changes. I am not eager about rewriting the crypto module now 🙂  Only need to fix the bug.

Reply:
hcryptMsg_SRT_ParseMsg uses a pointer to `_hcMsg_SRT_MsgInfo`, while `_hcMsg_SRT_MsgInfo` has a pointer `.parseMsg    = hcryptMsg_SRT_ParseMsg`.

```c
static const hcrypt_MsgInfo _hcMsg_SRT_MsgInfo = {
	.hdr_len     = HCRYPT_MSG_SRT_HDR_SZ,
	.pfx_len     = HCRYPT_MSG_SRT_PFX_SZ,
	.getKeyFlags = hcryptMsg_SRT_GetKeyFlags,
	.getPki      = hcryptMsg_SRT_GetPki,
	.setPki      = hcryptMsg_SRT_SetPki,
	.resetCache  = hcryptMsg_SRT_ResetCache,
	.indexMsg    = hcryptMsg_SRT_IndexMsg,
	.parseMsg    = hcryptMsg_SRT_ParseMsg
};

const hcrypt_MsgInfo* hcryptMsg_SRT_InitMsgInfo(void)
{
	return (&_hcMsg_SRT_MsgInfo);
}
```